src/App.jsx

```diff
@@ -3,26 +3,96 @@ import './App.scss';
 import './styles/general.scss';
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
+import { getTodos, getUser } from './api/api';
 
 class App extends React.Component {
   state = {
     todos: [],
     selectedUserId: 0,
+    title: '',
+    status: 'All',
   };
 
+  componentDidMount() {
+    getTodos()
+      .then(todos => this.setState({
+        todos: todos.filter(todo => todo.title && todo.userId),
+      }));
+  }
+
+  setUserId = (id) => {
+    this.setState({ selectedUserId: id });
+  }
+
+  setTitle = (title) => {
+    this.setState({ title });
+  }
+
+  setStatus = (status) => {
+    this.setState({ status });
+  }
+
+  clearUser = () => {
+    this.setState({
+      selectedUserId: '',
+    });
+  }
+
+  changeStatus = (id) => {
+    this.setState(state => ({
+      todos: state.todos.map((todo) => {
+        if (todo.id === id) {
+          return {
+            ...todo,
+            completed: !todo.completed,
+          };
+        }
+
+        return { ...todo };
+      }),
+    }));
+  }
+
   render() {
-    const { todos, selectedUserId } = this.state;
+    const { todos, selectedUserId, title, status } = this.state;
+
+    const todosByTitle = todos.filter(todo => todo.title.includes(title));
+
+    let visibleTodos;
+
+    switch (status) {
+      case 'completed':
+        visibleTodos = todosByTitle.filter(todo => todo.completed);
+        break;
+
+      case 'active':
+        visibleTodos = todosByTitle.filter(todo => !todo.completed);
+        break;
+      default:
+        visibleTodos = todosByTitle;
+        break;
+    }
```

Comment:
```suggestion
    const todosByTitle = todos.filter(todo => todo.title.includes(title));

    switch (status) {
      case 'completed':
        todosByTitle = todosByTitle.filter(todo => todo.completed);
        break;

      case 'active':
        todosByTitle = todosByTitle.filter(todo => !todo.completed);
        break;
      default:
        break;
    }
```
this way should work too